snuba/clickhouse/optimize/optimize.py

```diff
@@ -267,6 +278,22 @@ def get_current_large_merges(
     return merge_info
 
 
+def _get_partitions_to_optimize(
+    schedule: OptimizationSchedule,
+) -> deque[str]:
+    partitions_to_optimize: deque[str] = deque()
+
+    max_partition_group_size = max(
+        len(partition_group) for partition_group in schedule.partitions_groups
+    )
+
+    for i in range(max_partition_group_size):
+        for partition_group in schedule.partitions_groups:
+            if i < len(partition_group):
+                partitions_to_optimize.append(partition_group[i])
+    return partitions_to_optimize
```

Comment:
I went with approach 1, but the reason why I was zig-zaging instead of doing approach 2 is because i was trying to preserve [the order in which partitions are optimized](https://github.com/getsentry/snuba/blob/cc6f37c4513f1d5d2356d0e84033a94ccb08ee3e/snuba/clickhouse/optimize/optimize_scheduler.py#L77-L79)

My understanding is that if we had the `schedule.partitions_groups = [[1,2], [3,4,5], [6,7]]`, then before this PR, the order in which partitions should be optimized is
- partition 1, 3, 6 (simultaneously optimized)
- partition 2, 4, 7
- partition 5

so by zig-zaging, `partitions_to_optimize` becomes `[1,3,6,2,4,7,5]`